### PR TITLE
[Ibis] Call preparation pass -- part 1

### PR DESCRIPTION
**Pull request by teqdruid**
Lowering the Ibis control flow starts with converting all calls to the DC dialect's values with one struct representing the packed arguments.

In the interest of having small PRs, this only implements narrowing the arguments to one struct. Part 2 will then wrap them in DC values.

### CODE REVIEW

File: lib/Dialect/Ibis/Transforms/IbisCallPrep.cpp [hunk 1]
```diff
@@ -0,0 +1,270 @@
...
+  return SymbolRefAttr::get(cls.getSymNameAttr(),
+                            {FlatSymbolRefAttr::get(path.getLeafReference())});
+}
+
+namespace {
+/// For each CallOp, the corresponding method signature will have changed. Pack
+/// all the operands into a struct.
+struct MergeCallArgs : public OpConversionPattern<CallOp> {
+  MergeCallArgs(MLIRContext *ctxt, const CallPrepPrecomputed &info)
+      : OpConversionPattern(ctxt, 1), info(info) {}
```
Comment (mortbopet) [inline lib/Dialect/Ibis/Transforms/IbisCallPrep.cpp:133]: Why give this (and the pattern below) an explicit benefit value and not just go with whatever's default? (`OpConversionPattern(ctxt)`).

**Review by mortbopet — APPROVED** [review]
LGTM. To make sure we're going down the right path here, i'd really like to see a piece of high-level IBIS IR containing non-trivial logic which we expect will be the output of the AST conversion.

**Review by blakep-msft — COMMENTED** [review]
LGTM

File: lib/Dialect/Ibis/Transforms/IbisCallPrep.cpp (hunk 1 above)
Comment (teqdruid) [reply lib/Dialect/Ibis/Transforms/IbisCallPrep.cpp]: VSCode was insisting I needed it. But it was wrong.

**Comment by teqdruid** [comment]
Lowering the Ibis control flow starts with converting all calls to the DC dialect's values with one struct representing the packed arguments.

In the interest of incremental development, this only implements narrowing the arguments to one struct. Part 2 will then wrap them in DC values.

**Pull request merged**